packages/reports/addon/consumers/request/sort.js

```diff
@@ -46,16 +46,42 @@ export default ActionConsumer.extend({
 
       request.removeSortByMetricName(metricName);
     },
+    /**
+     * @action REMOVE_SORT_WITH_PARAM
+     * @param {Object} route - route that has a model that contains a request property
+     * @param {Object} metric - metadata model of metric to remove from sort
+     * @param {Object} [parameters] - optional metric parameters
+     */
+    [RequestActions.REMOVE_SORT_WITH_PARAM]({ currentModel }, metric, parameters) {
+      let request = get(currentModel, 'request');
+
+      request.removeSortMetricByModelAndParam(metric, parameters);
+    },
     /**
      * @action REMOVE_METRIC
      * @param {Object} route - route that has a model that contains a request property
      * @param {Object} metric - metadata model of metric to remove
      */
     [RequestActions.REMOVE_METRIC](route, metric) {
       // Find and remove any `sorts` attached to the metric
-      let metricName = get(metric, 'name');
+      if (get(metric, 'hasParameters')) {
+        // מot passing parameters to remove sorts with any parameter
+        get(this, 'requestActionDispatcher').dispatch(RequestActions.REMOVE_SORT_WITH_PARAM, route, metric);
+      } else {
+        let metricName = get(metric, 'name');
 
-      get(this, 'requestActionDispatcher').dispatch(RequestActions.REMOVE_SORT, route, metricName);
+        get(this, 'requestActionDispatcher').dispatch(RequestActions.REMOVE_SORT, route, metricName);
+      }
+    },
+    /**
+     * @action REMOVE_METRIC_WITH_PARAM
+     * @param {Object} route - route that has a model that contains a request property
+     * @param {Object} metric - metadata model of metric to remove
+     * @param {Object} parameters - metric parameters
+     */
+    [RequestActions.REMOVE_METRIC_WITH_PARAM](route, metric, parameters) {
+      // Find and remove any `sorts` attached to the metric and parameters
+      get(this, 'requestActionDispatcher').dispatch(RequestActions.REMOVE_SORT_WITH_PARAM, route, metric, parameters);
```

Comment:
If all this action does is dispatch another action, isn't this action just a middle man?

Reply:
Nevermind, I see what's going on